obd/decoders.py

```diff
@@ -484,3 +489,31 @@ def monitor(messages):
             mon.add_test(test)
 
     return mon
+
+
+def encoded_string(length):
+    """ Extract an encoded string from multi-part messages """
+    return functools.partial(decode_encoded_string, length=length)
+
+
+def decode_encoded_string(messages, length):
+    d = messages[0].data[2:]
+
+    # If a bogus string length is returned, discard it
+    if len(d) < length:
+        logger.debug("Invalid string returned. Discarding...")
```

Comment:
Is it worth printing the sting here? That way the user can see what was bogus, that should help in debugging

Reply:
We could, but I'm not sure how useful this would be. Multiple PIDs use this decoder with different string formats (e.g. VIN as an ASCII-string, CVN as a hex string, etc.). We could pass in a format string to aid with correctly logging the string output, but this then leads to an uglier and more fragile interface for what is ultimately a corner case.

Reply:
There are lots of different car models, so I think corner cases come up somewhat often. When they do having the string in the log should help debug the issue and will at least improve the quality of the log dumped in a GitHub issue.